src/node/stage/approved.rs

```diff
@@ -1611,45 +1642,69 @@ impl Approved {
     }
 
     fn update_our_section(&mut self, core: &mut Core, details: SectionUpdateDetails) -> Result<()> {
-        trace!("Update our Section with {:?}", details);
+        trace!("update our section with {:?}", details);
+        info!("handle SectionInfo: {:?}", details.our.info.value);
+
         let old_prefix = *self.shared_state.our_prefix();
         let was_elder = self.is_our_elder(core.id());
         let sibling_prefix = details.sibling.as_ref().map(|sibling| sibling.key.value.0);
 
-        self.add_force_gossip_peer(&details.our.info.value, old_prefix, was_elder);
+        let mut old_shared_state = self.shared_state.clone();
 
-        self.update_our_key_and_info(core, details.our.key, details.our.info)?;
+        let promoted_nodes: Vec<_> = details
+            .all_elders()
+            .filter(|p2p_node| {
+                !self
+                    .shared_state
+                    .our_info()
+                    .elders
+                    .contains_key(p2p_node.name())
+            })
+            .cloned()
+            .collect();
+
+        self.add_force_gossip_peer(&details.our.info.value, old_prefix, was_elder);
+        self.update_our_key_and_info(core, details.our.key, details.our.info.clone())?;
 
         if let Some(sibling) = details.sibling {
             self.add_force_gossip_peer(&sibling.info.value, old_prefix, was_elder);
             self.shared_state.sections.update_keys(sibling.key);
-            self.update_neighbour_info(core, sibling.info);
-        }
+            self.update_neighbour_info(core, sibling.info.clone());
 
-        let elders_info = self.shared_state.our_info();
-        let new_prefix = elders_info.prefix;
-        let is_elder = elders_info.elders.contains_key(core.name());
-
-        core.msg_filter.reset();
-
-        if was_elder || is_elder {
-            info!("handle SectionInfo: {:?}", elders_info);
-        } else {
-            trace!("unhandled SectionInfo");
-            return Ok(());
+            old_shared_state.update_our_section(sibling.info, sibling.sibling_our_key);
+            old_shared_state.sections.add_neighbour(details.our.info);
+            old_shared_state
+                .sections
+                .update_keys(sibling.sibling_their_key);
         }
 
-        self.section_update_barrier = Default::default();
-
+        let new_prefix = &self.shared_state.our_info().prefix;
         if new_prefix.is_extension_of(&old_prefix) {
             info!("Split");
-        } else if old_prefix.is_extension_of(&new_prefix) {
+
+            self.send_promote(
+                core,
+                old_shared_state,
+                self.consensus_engine.parsec_version() + 1,
+                promoted_nodes.clone(),
+            )?;
```

Comment:
Shouldn't we send this only to the nodes from the sibling section?

Reply:
the receiver need to carry out a check to `ignore` the incoming message if it's not in the sibling sectionm anyway.
Didn't carry out filtering here, on sender's side, is mainly to simplify the coding, as there is another `send_promote` call immediately below.

I will consider adding a check inside the send_promote function to see if works.

Reply:
addressed by PR https://github.com/maidsafe/routing/pull/2166